R/zchunk_L111.nonghg_en_R_S_T_Y.R

```diff
@@ -1,110 +1,303 @@
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) ->
+      EDGAR_NH3
+
+    # First, add gas name and bind all dataframes together.
+
+    L101.so2_tgej_USA_en_Sepa_F_Yh$Non.CO2 <- "SO2"
+    L101.co_tgej_USA_en_Sepa_F_Yh$Non.CO2 <- "CO"
+    L101.nox_tgej_USA_en_Sepa_F_Yh$Non.CO2 <- "NOx"
+    L101.voc_tgej_USA_en_Sepa_F_Yh$Non.CO2 <- "NMVOC"
+
+    L101.nh3_tgej_USA_en_Sepa_F_Yh %>%
+      mutate(Non.CO2 = "NH3") %>%
+      bind_rows(L101.so2_tgej_USA_en_Sepa_F_Yh, L101.co_tgej_USA_en_Sepa_F_Yh,
+                L101.nox_tgej_USA_en_Sepa_F_Yh, L101.voc_tgej_USA_en_Sepa_F_Yh, .)  %>%
+      group_by(Non.CO2, sector, fuel) %>%
+      # Rename columns to match other tables.
+      rename(EPA_agg_sector = sector) %>%
+      rename(EPA_agg_fuel = fuel) ->
+      L111.nonghg_tgej_USA_en_Sepa_F_Yh.tmp1
+
+    # Compute unscaled emissions by country and technology.
+
+    # Add non-ghg gases for combination of sector, fuel, and technology.
+    L101.in_EJ_R_en_Si_F_Yh %>%
+      # Renmae columns for joining.
+      rename(xyear = year, energy = value) %>%
+      left_join_keep_first_only(select(GCAM_sector_tech, sector, fuel, technology, EPA_agg_sector, EPA_agg_fuel),
+                                  by = c("sector", "fuel", "technology"))  %>%
+      repeat_add_columns(tibble(Non.CO2 = emissions.NONGHG_GASES)) ->
+      L111.nonghg_tg_R_en_Si_F_Yh.tmp1
+
+    # Match in emissions factors and then compute unscaled emissions.
+
+    # Aggregate by EDGAR sector and region
+    L111.nonghg_tg_R_en_Si_F_Yh.tmp1 %>%
+      rename(year = xyear) %>%
+      # Join in USA non-ghg emission factor data.  Keep NAs for now.
+      left_join(L111.nonghg_tgej_USA_en_Sepa_F_Yh.tmp1, by = c("Non.CO2", "EPA_agg_sector", "EPA_agg_fuel", "year"))  %>%
+      rename(emfact = value)  %>%
+      # Scale emission factors by energy sectors.
+      mutate(epa_emissions = energy * emfact) %>%
+      na.omit() %>%
+      left_join_keep_first_only(select(GCAM_sector_tech, EDGAR_agg_sector, sector, fuel),
+                                by = c("sector", "fuel")) ->
+      L111.nonghg_tg_R_en_Si_F_Yh.tmp1
+
+    # Create column of total EPA emissions by EDGAR sector and region
+    L111.nonghg_tg_R_en_Si_F_Yh.tmp1 %>%
+      group_by(GCAM_region_ID, Non.CO2, EDGAR_agg_sector, year) %>%
+      summarise(EPA_emissions = sum(epa_emissions)) ->
+      L111.nonghg_tg_R_en_Sedgar_Yh.tmp1
+
+    # Compute EDGAR emissions by region and sector.
+
+    # Add gas name and bind all dataframes together.
+    EDGAR_SO2$Non.CO2 <- "SO2"
+    EDGAR_CO$Non.CO2 <- "CO"
+    EDGAR_NOx$Non.CO2 <- "NOx"
+    EDGAR_NH3$Non.CO2 <- "NH3"
+
+    EDGAR_NMVOC %>%
+      mutate(Non.CO2 = "NMVOC") %>%
+      bind_rows(EDGAR_SO2, EDGAR_CO, EDGAR_NOx, ., EDGAR_NH3)  %>%
+      # Filter for EDGAR years (1971-2008) and 1970.
+      filter(year, year %in% emissions.EDGAR_YEARS_PLUS)  %>%
+      left_join(EDGAR_sector, by =c("IPCC", "IPCC_description"))  %>%
```

Comment:
I can't do the `left_join_error_no_match` because the # of rows is not equal?
```Error in left_join_error_no_match(., EDGAR_sector, by = c("IPCC", "IPCC_description")) : 
  left_join_no_match: NA values in new data columns```
I can do the space after `=`